3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/IabEnrichment.scala

```diff
@@ -0,0 +1,265 @@
+   *
+   * @param config    The iab_spiders_and_robots_enrichment JSON
+   * @param schemaKey The SchemaKey provided for the enrichment
+   *                  Must be a supported SchemaKey for this enrichment
+   * @param localMode Whether to use the local IAB database file
+   *                  Enabled for tests
+   * @return a configured IabEnrichment instance
+   */
+  def parse(config: JValue, schemaKey: SchemaKey, localMode: Boolean): ValidatedNelMessage[IabEnrichment] =
+    isParseable(config, schemaKey).flatMap { conf =>
+      def uri(name: String) = getUriTupleFromName(conf, name).sequenceU
+
+      def string(name: String) = getStringFromName(conf, name).sequenceU
+
+      (uri("ipFile")               |@| uri("excludeUseragentFile") |@| uri("includeUseragentFile")
+        |@| string("httpUsername") |@| string("httpPassword")) {
+        IabEnrichment(_, _, _, _, _, localMode)
+      }
+    }
+
+  /**
+   * Creates (URI, String) tuples used in the IabEnrichment case class.
+   *
+   * @param config The iab_spiders_and_robots_enrichment JSON
+   * @param name   The name of the field.
+   *             e.g. "ipFile", "excluseUseragentFile", "includeUseragentFile"
+   * @return None if the field does not exist,
+   *         Some(Failure) if the URI is invalid,
+   *         Some(Success) if it is found
+   */
+  private def getUriTupleFromName(config: JValue, name: String): Option[ValidatedNelMessage[(String, URI, String)]] =
+    if (ScalazJson4sUtils.fieldExists(config, "parameters", name)) {
+      val uri = ScalazJson4sUtils.extract[String](config, "parameters", name, "uri")
+      val db  = ScalazJson4sUtils.extract[String](config, "parameters", name, "database")
+
+      (uri.toValidationNel |@| db.toValidationNel) { (uri, db) =>
+        for {
+          u <- getDatabaseUri(uri, db).toValidationNel: ValidatedNelMessage[URI]
+        } yield (name, u, db)
+
+      }.flatMap(x => x).some
+
+    } else None
+
+  /**
+   * Extracts simple string fields from an enrichment JSON.
+   *
+   * @param config The iab_spiders_and_robots_enrichment JSON
+   * @param name   The name of the lookup:
+   *               "geo", "isp", "organization", "domain"
+   * @return None if the field does not exist,
+   *         Some(Success) if it is found
+   */
+  private def getStringFromName(config: JValue, name: String): Option[ValidatedNelMessage[String]] =
+    if (ScalazJson4sUtils.fieldExists(config, "parameters", name)) {
+      ScalazJson4sUtils.extract[String](config, "parameters", name).toValidationNel.some
+    } else None
+
+  /**
+   * Convert the path to the IAB file from a
+   * String to a Validation[URI].
+   *
+   * @param uri      URI to the IAB database file
+   * @param database Name of the IAB database
+   * @return a Validation-boxed URI
+   */
+  private def getDatabaseUri(uri: String, database: String): ValidatedMessage[URI] =
+    ConversionUtils
+      .stringToUri(uri + "/" + database)
```

Comment:
I would shy away from doing path handling using string concatenation, what if `uri` has a `/` at the end? does `ConversionUtils` handle it?

Reply:
It should work with trailing slashes: https://github.com/snowplow/snowplow/blob/master/3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/utils/conversionUtilsSpecs.scala#L73

Reply:
this example doesn't have two trailing slashes afaict

Reply:
Seems to work fine (the multiple slashes are simply added to the URI with no errors). I've added a normalize call here just in case - that should remove the extra slashes.

Reply:
normalize doesn't deal with the case I meant, doing string concatenation on paths can result in:

```scala
val uri = "http://google.com/"
val databse = "database"
uri + "/" + database
// http://google.com//database

ConversionUtils.stringToUri(uri + "/" + database)
// Success(Some(http://google.com//database))

new URI("http://google.com/").resolve("/" + database)
// http://google.com/database
```

Do you see what I mean now?